controllers/nodehealthcheck_controller.go

```diff
@@ -583,7 +580,8 @@ func (r *NodeHealthCheckReconciler) remediate(ctx context.Context, node *v1.Node
 				log.Info("skipping timeout annotation on remediation CR: Succeeded condition is True", "CR name", remediationCR.GetName())
 			}
 			startedRemediation := resources.FindStatusRemediation(node, nhc, func(r *remediationv1alpha1.Remediation) bool {
-				return r.Resource.GroupVersionKind() == remediationCR.GroupVersionKind()
+				return r.Resource.GroupVersionKind() == remediationCR.GroupVersionKind() &&
```

Comment:
fixes ` timeout annotation is not added to a remediation CR, when it belongs to a later remediation with same kind, and lease expired while it's processed`